doc/workshops/plugin-development.rst

```diff
@@ -2,4 +2,1275 @@
+
+.. code-block:: python
+
+   """
+   Spyder Pomodoro Timer Preferences Page.
+   """
+   from qtpy.QtWidgets import QGridLayout, QGroupBox, QVBoxLayout
+   from spyder.api.preferences import PluginConfigPage
+   from spyder.api.translations import get_translation
+
+   from spyder_pomodoro_timer.spyder.config import POMODORO_DEFAULT
+
+   _ = get_translation("spyder_pomodoro_timer.spyder")
+
+
+   class SpyderPomodoroTimerConfigPage(PluginConfigPage):
+
+       # --- PluginConfigPage API
+       # ------------------------------------------------------------------------
+       def setup_page(self):
+           limits_group = QGroupBox(_("Time limits"))
+           pomodoro_spin = self.create_spinbox(
+               _("Pomodoro timer limit"),
+               _("min"),
+               "pomodoro_limit",
+               default=POMODORO_DEFAULT,
+               min_=5,
+               max_=100,
+               step=1,
+           )
+
+           pt_limits_layout = QGridLayout()
+           pt_limits_layout.addWidget(pomodoro_spin.plabel, 0, 0)
+           pt_limits_layout.addWidget(pomodoro_spin.spinbox, 0, 1)
+           pt_limits_layout.addWidget(pomodoro_spin.slabel, 0, 2)
+           pt_limits_layout.setColumnStretch(1, 100)
+           limits_group.setLayout(pt_limits_layout)
+
+           vlayout = QVBoxLayout()
+           vlayout.addWidget(limits_group)
+           vlayout.addStretch(1)
+           self.setLayout(vlayout)
+
+This corresponds to the usual code for user interfaces based on Qt widgets, in this case the section of our parameters corresponds to a ``QGroupBox``, where the parameters are organized vertically using a ``QVBoxLayout``, and each parameter corresponds to a ``QGridLayout`` where Labels and inputs are distributed (in this case a ``QSpinBox``).
+
+Spyder provides some helper functions to facilitate the work, for instance, ``create_spinbox`` which allows to instantiate and initialize in a single step the widgets corresponding preffix an suffix labels together with the spinbox.
+
+Propagate configuration changes
+-------------------------------
+
+Since we moved all the configuration information to ``conf.py``, now we have to import it from there into ``widgets.py``.
+
+`v0.0.4dev-v0.0.5dev widgets.py diff`_
+
+.. _v0.0.4dev-v0.0.5dev widgets.py diff: https://github.com/map0logo/spyder-pomodoro-timer/compare/v0.0.4-dev0...v0.0.5-dev0#diff-f156d182eb57c9fa20d1ca4b9677d13b158447c04f90724d936bb77a50df31bc
+
+.. code-block:: python
+
+   # Local imports
+   from spyder_pomodoro_timer.spyder.config import (
+       CONF_SECTION,
+       CONF_DEFAULTS,
+       CONF_VERSION,
+   )
+
+Once we have the form ready, we can access the configuration value from anywhere in our plugin using the ``get_conf`` method, in this case we use it to access the value of the configurable parameter ``pomodoro_limit`` from the configuration instead of the constant ``POMODORO_DEFAULT``.
```

Comment:
```suggestion
Now we can access the configuration options from anywhere in our plugin using the ``get_conf`` method. In this case we use it to access the value of ``pomodoro_limit`` from the configuration instead of the constant ``POMODORO_DEFAULT``.
```